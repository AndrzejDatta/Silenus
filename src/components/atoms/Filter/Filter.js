import React from "react";
import PropTypes from "prop-types";
import { StyledFilter, StyledFilterText } from "./Filter.styles";
import Icon from "components/atoms/icon/icon.component";

const Filter = ({
  text,
  icon,
  isBigFontSize,
  color,
  isCategoryInIdentifyPage,
}) => {
  return (
    <StyledFilter isCategoryInIdentifyPage={isCategoryInIdentifyPage}>
      {icon && <Icon icon={icon} size="tiny" color="green" />}
      <StyledFilterText
        isCategoryInIdentifyPage={isCategoryInIdentifyPage}
        isBigFontSize={isBigFontSize}
        color={color}
      >
        {text}
      </StyledFilterText>
    </StyledFilter>
  );
};

Filter.defaultProps = {
  text: "Filter",
  icon: "",
  isBigFontSize: false,
  color: "black",
  isCategoryInIdentifyPage: false,
};

Filter.propTypes = {
  /**text inject in div */
  text: PropTypes.string.isRequired,
  /**src to icon */
  icon: PropTypes.string,
  /**Size of text */
  isBigFontSize: PropTypes.bool,
  /**Color of text */
  color: PropTypes.string,
  /**bool which decides about padding */
  isCategoryInIdentifyPage: PropTypes.bool,
};

export default Filter;
