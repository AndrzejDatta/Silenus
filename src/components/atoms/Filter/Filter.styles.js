import styled from "styled-components";

export const StyledFilter = styled.button`
  background-color: ${({ theme }) => theme.color.lightGrey};
  padding: 1px;
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  align-items: center;
  border: none;
  border-radius: 8px;
  min-width: 112px;
  margin: 0px 5px;
  max-width: 140px;
  ${({ isCategory }) => isCategory && `padding:5px 10px`}
`;

export const StyledFilterText = styled.p`
  font-size: ${(props) =>
    props.isBig
      ? ({ theme }) => theme.font.size.big
      : ({ theme }) => theme.font.size.medium};
  color: ${(props) =>
    props.isDashboard ? ({ theme }) => theme.color.fontGreen : "black"};
  font-family: ${({ theme }) => theme.font.family.poppins};
`;
