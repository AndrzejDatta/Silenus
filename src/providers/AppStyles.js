import { ThemeProvider } from 'styled-components';
import { theme } from 'assets/styles/theme';
import { GlobalStyle } from 'assets/styles/GlobalStyles';
export const AppStyles = ({ children }) => {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  );
};
