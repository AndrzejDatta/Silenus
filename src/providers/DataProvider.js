import { createContext, useState } from "react";
import { dataPL } from "assets/data/data";

export const dataContext = createContext({
  startPage: {},
  loginPage: {},
  Dashboard: {},
  MojeDoniczki: {},
  DodajDoniczke: {},
  MojeRośliny: {},
  KalendarzOpieki: {},
  KontoUżytkownika: {},
  ZidentyfikujRośline: {},
  PotDetails: {},
  PlantDetails: {},
  Navbar: {},
  images: {},
  navIcons: {},
  potIcons: {},
  categoryIcons: {},
  navIconsColor: [],
  setNavIconColor: () => {},
  changeColor: () => {},
  importAll: () => {},
});

const objectName = [
  "startPage",
  "loginPage",
  "Dashboard",
  "MojeDoniczki",
  "DodajDoniczke",
  "MojeRośliny",
  "KalendarzOpieki",
  "KontoUżytkownika",
  "ZidentyfikujRośline",
  "PotDetails",
  "PlantDetails",
  "Navbar",
];

export const DataProvider = ({ children }) => {
  for (let i = 0; i < dataPL.length; i++) {
    objectName[i] = { ...dataPL[i] };
  }
  const [
    startPage,
    loginPage,
    Dashboard,
    MojeDoniczki,
    DodajDoniczke,
    MojeRośliny,
    KalendarzOpieki,
    KontoUżytkownika,
    ZidentyfikujRośline,
    PotDetails,
    PlantDetails,
    Navbar,
  ] = objectName;

  const [navIconsColor, setNavIconColor] = useState([
    {
      name: "plant",
      src: "potted_plant.svg",
      isClicked: false,
    },
    {
      name: "calendar",
      src: "calendar.svg",
      isClicked: false,
    },
    {
      name: "camera",
      src: "camera.svg",
      isClicked: false,
    },
    {
      name: "user",
      src: "accountCircle.svg",
      isClicked: false,
    },
  ]);

  function importAll(r) {
    let images = {};
    r.keys().map((item, index) => {
      images[item.replace("./", "")] = r(item);
    });
    return images;
  }

  const images = importAll(
    require.context(
      "assets/icons/userAccountOption",
      false,
      /\.(png|jpe?g|svg)$/
    )
  );
  const navIcons = importAll(
    require.context("assets/icons/navIcons", false, /\.(png|jpe?g|svg)$/)
  );

  const potIcons = importAll(
    require.context("assets/icons/potStatus", false, /\.(png|jpe?g|svg)$/)
  );

  const categoryIcons = importAll(
    require.context("assets/icons/categoryIcons", false, /\.(png|jpe?g|svg)$/)
  );

  function changeColor(nameElement) {
    setNavIconColor((prevData) =>
      prevData.map((item) => {
        if (item.name === nameElement && !item.isClicked) {
          return {
            ...item,
            isClicked: true,
          };
        } else {
          return {
            ...item,
            isClicked: false,
          };
        }
      })
    );
  }

  return (
    <dataContext.Provider
      value={{
        startPage,
        loginPage,
        Dashboard,
        DodajDoniczke,
        MojeDoniczki,
        MojeRośliny,
        KalendarzOpieki,
        KontoUżytkownika,
        ZidentyfikujRośline,
        PotDetails,
        PlantDetails,
        Navbar,
        images,
        navIcons,
        potIcons,
        categoryIcons,
        navIconsColor,
        setNavIconColor,
        changeColor,
        importAll,
      }}
    >
      {children}
    </dataContext.Provider>
  );
};

//TODO:improve way
