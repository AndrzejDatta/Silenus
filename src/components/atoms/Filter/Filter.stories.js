import React from "react";
import Filter from "./Filter";
import { storiesName } from "helpers/storiesName";

export default {
  title: `${storiesName.atoms}Filter`,
  component: Filter,
  argTypes: {
    isBigFontSize: {
      options: [true, false],
      control: { type: "radio" },
    },
    isCategoryInIdentifyPage: {
      options: [true, false],
      control: { type: "radio" },
    },
  },
};

const Template = (args) => <Filter {...args} />;

export const Default = Template.bind({});

Default.args = {
  isBigFontSize: true,
  text: "Filter",
  color: "black",
  isCategoryInIdentifyPage: false,
};
