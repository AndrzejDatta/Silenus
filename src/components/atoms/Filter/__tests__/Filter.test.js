import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import { AppStyles } from "providers/AppStyles";
import { render, screen } from "@testing-library/react";
import Filter from "components/atoms/Filter/Filter";

describe("Test Filter Component", () => {
  it("Test default style", () => {
    const tree = renderer
      .create(
        <AppStyles>
          <Filter />
        </AppStyles>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Test filter in UserActivity Section in Dashboard", () => {
    render(
      <AppStyles>
        <Filter isBigFontSize={true} color="#3AA688" />
      </AppStyles>
    );
    expect(screen.getByText("Filter")).toHaveStyleRule("font-size", "16px");
    expect(screen.getByText("Filter")).toHaveStyleRule("color", "#3AA688");
  });
});
