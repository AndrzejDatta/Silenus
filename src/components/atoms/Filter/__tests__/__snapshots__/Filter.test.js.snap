// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Filter Component Test default style 1`] = `
.c0 {
  background-color: #F9F9F9;
  padding: 1px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  border-radius: 8px;
  min-width: 112px;
  margin: 0px 5px;
  max-width: 140px;
}

.c1 {
  font-size: 14px;
  color: black;
  font-family: "Poppins",sans-serif;
}

<button
  className="c0"
>
  <p
    className="c1"
    color="black"
  >
    Filter
  </p>
</button>
`;
